### DISCUSSION TITLE
Flash failure Orin NX

### DISCUSSION THREAD

**Discussion by LorenzoRuffati** [OE4T/meta-tegra#1249] [Q&A] [answered]
Hi, I'm encountering a problem when trying to flash an image generated using the demo distro (the build completes without any issues)

These are the last lines of the flash messages, does anyone know what's causing this/how to fix it?

```
 Entering RCM boot

[   0.0348 ] mb1_t234_prod_aligned_sigheader.bin.encrypt filename is from --mb1_bin
[   0.0348 ] psc_bl1_t234_prod_aligned_sigheader.bin.encrypt filename is from --psc_bl1_bin
[   0.0348 ] rcm boot with presigned binaries
[   0.0356 ] tegrarcm_v2 --new_session --chip 0x23 0 --uid --download bct_br br_bct_BR.bct --download mb1 mb1_t234_prod_aligned_sigheader.bin.encrypt --download psc_bl1 psc_bl1_t234_prod_aligned_sigheader.bin.encrypt --download bct_mb1 mb1_bct_MB1_sigheader.bct.encrypt
Error: Return value 8
Command tegrarcm_v2 --new_session --chip 0x23 0 --uid --download bct_br br_bct_BR.bct --download mb1 mb1_t234_prod_aligned_sigheader.bin.encrypt --download psc_bl1 psc_bl1_t234_prod_aligned_sigheader.bin.encrypt --download bct_mb1 mb1_bct_MB1_sigheader.bct.encrypt
```

Thanks for the help

**Comment by ichergui** [comment]
Hey @LorenzoRuffati 
Welcome the OE4T community.
Could you please share the steps you took for the build ? 
Are you using a custom carrier board ? or the `Xavier NX` carrier board ?
Do you have secure boot enable ?

Thanks 
Ilies

**Reply by LorenzoRuffati** [reply]
Hi, I built it by following the steps in the README of tegra-demo-distro

The machine I used was "p3509-a02-p3767-0000" 

I'm using this carrier board: https://www.waveshare.com/jetson-orin-nx-16g-dev-kit.htm

If I understand correctly it should be the same as the Xavier NX carrier board

No, secure boot is disabled in the boot settings

**Reply by ichergui** [reply]
I need to check the carrier board. 
I'm didn't try that machine yet, I just receive the module. I will keep you posted.
@madisongh Could you please share your thoughts ?

**Reply by dwalkes** [reply]
@LorenzoRuffati did you see the thread at https://github.com/OE4T/meta-tegra/discussions/1217 ? Looks like a very similar error reported there.  You might want to follow the path suggested there for troubleshooting.  Comparing stock L4T flash results with OE4T will likely be helpful to diagnose.  Another thread with this error can be found at https://github.com/OE4T/meta-tegra/discussions/1138

**Reply by LorenzoRuffati** [reply]
Thanks, I'll check them out

**Comment by LorenzoRuffati** [comment]
I have fixed the previous issue (I was using a VM and the tool would not work properly in that setup) but I think the flash is still not succeeding

I'm executing `initrd-flash` in order to flash the board since it uses an nvme drive as the boot drive, the flash command executes and I see from the serial monitor as well as the command output that it is flashing the images. However when it starts flashing the ext4 rootfs image it just hangs with no output nor logs on the serial

Is there a way to verify if something is hanging or it's just the image being large and requiring more time to flash (though I've left it flashing for a significant amount of time)

**Reply by madisongh** [reply]
Do you have the `bmaptool` command available?  If not, I'd strongly recommend installing the `bmap-tools` package do so.  Flashing the rootfs onto a large NVME drive would be excruciatingly slow using `dd`.  (The initrd-flash script automatically uses `bmaptool` if available, with `dd` as a fallback.)

Are there any kernel logs on your host machine from around that time?   What does `top` report?

What is your host OS?

**Reply by LorenzoRuffati** [reply, marked as answer]
Hi, thanks for the hint about bmaptool, I'll install it.

I think I had done something wrong when building the first time because I tried again with a fresh build and it was slow but successful

My host OS is ubuntu 20.04, I'll keep in mind to check the kernel logs and top should any other issue arise
